pythonsd/tests/__init__.py

```diff
@@ -131,6 +132,88 @@ def test_api_error(self):
         self.assertContains(response, "There are no upcoming events")
 
 
+class TestYouTubeRecentVideosView(test.TestCase):
+    def setUp(self):
+        self.url = reverse("recent_videos")
+
+        fp = os.path.join(os.path.dirname(__file__), "data/youtube-video-feed.xml")
+        with open(fp) as fd:
+            self.api_response = fd.read()
+
+        self.expected_videos = [
+            {
+                # These are actually datetime.datetimes
+                "datetime": "2023-01-27T17:52:55Z",
+                "id": "MvpiCyPpAhM",
+                "title": "San Diego Python Monthly Meetup January 2023",
+                "url": "https://www.youtube.com/watch?v=MvpiCyPpAhM",
+            },
+            {
+                "datetime": "2022-12-22T13:55:08Z",
+                "id": "qyfmJVBZFIQ",
+                "title": "San Diego Python Monthly Meetup December 2022",
+                "url": "https://www.youtube.com/watch?v=qyfmJVBZFIQ",
+            },
+        ]
+
+    def tearDown(self):
+        super().tearDown()
+        cache.clear()
+
+    def test_no_videos(self):
+        with mock.patch(
+            "pythonsd.views.RecentVideosView.get_recent_videos", return_value=[]
+        ) as mock_get:
+            response = self.client.get(self.url)
+            self.assertContains(response, "Check out our")
+
+    def test_preloaded_videos(self):
+        with mock.patch(
+            "pythonsd.views.RecentVideosView.get_recent_videos",
+            return_value=self.expected_videos,
+        ) as mock_get:
+            response = self.client.get(self.url)
+            self.assertContains(response, "MvpiCyPpAhM")
+            self.assertContains(response, "<iframe")
```

Comment:
I see that the widget is also being tested in `test_html_video_widget`. Is there a reason to check for the iframe twice.

Reply:
There isn't but it is a double check in case we change things